File: AbstractFactoryUsingBuilderGOlang/abstractFactory/VEGGYDealer.go
package abstractFactory

import (
	"myProject7/PizzaBuilder"
	"myProject7/SandwichBuilder"
)

type VEGGYDealer struct{}

func (V VEGGYDealer) CreatePizza() PizzaBuilder.Pizza {
	veggyBuilder := PizzaBuilder.GetBuilder("veggy")
	waitress := PizzaBuilder.NewWaitressPizza(veggyBuilder)
	veggyPizza := waitress.BuildPizza()

	return veggyPizza
}

func (V VEGGYDealer) CreateSandwich() SandwichBuilder.Sandwich {

	veggyBuilder := SandwichBuilder.GetBuilder("veggy")
	waitress := SandwichBuilder.NewWaitressSandwich(veggyBuilder)
	veggySandwich := waitress.BuildSandwich()

	return veggySandwich
}

File: AbstractFactoryUsingBuilder/AbstractFactory/AbstractFactory/VEGGYDealer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstractFactory
{
    public class VEGGYDealer : AbstractFactory
    {

        public override PIZZA createPIZZA()
        {
            PizzaBuilder builderVEGGYPIZZA = new BuilderVEGGYPIZZA();
            WAITRESSPIZZA wai = new WAITRESSPIZZA(builderVEGGYPIZZA);
            wai.constructPizza();
            
            return wai.getPizza();
        }

        public override SANDWICH createSANDWICH()
        {
            SandwichBuilder builderVEGGYSANDWICH = new BuilderVEGGYSANDWICH();
            WAITRESSSANDWICH wai = new WAITRESSSANDWICH(builderVEGGYSANDWICH);
            wai.constructSandwich();

            return wai.getSandwich();
        }

    }
}
